project/KernelBoiler.scala

```diff
@@ -150,7 +150,9 @@ object KernelBoiler {
                 |      def empty: ${`(A..N)`} = ${nullaryTuple("empty")}
                 |      def inverse(x: ${`(A..N)`}): ${`(A..N)`} = ${unaryTuple("inverse")}
                 |    }
-                |  implicit def catsKernelStdOrderForTuple${arity}[${`A..N`}](implicit ${constraints("Order")}): Order[${`(A..N)`}] =
+                |  implicit def catsKernelStdOrderForTuple${arity}[${`A..N`}](implicit ${constraints(
+                  "Order"
+                )}): Order[${`(A..N)`}] =
```

Comment:
@armanbilge, check out that the formatting breaks code block margins. I believe it should not work in this way. I would suppose it is a bug – the formatter either should be smarter preserving the block margins or don't attempt to format it at all.

Reply:
That's a good point. Out of curiosity, do these changes affect the output?

Reply:
1. line 146 above: definitely not new
2. in normal interpolated strings, everything between `${` and `}` is evaluated and interpolated, no newlines are material

Reply:
@kitbellew oh, I see what you're saying now. Indeed, there was formatting in interpolated strings already applied by one of the previous scalafmt versions. Seems that with this release scalafmt begins applying such a formatting more meticulously. More likely that it is not a new bug has been introduced but rather an older bug has been fixed with this release.

Although I still feel uncertain whether it was initially a good idea of adding line breaks inside pre-formatted blocks (even though solely inside interpolated strings). Is it configurable somehow?

Reply:
> Although I still feel uncertain whether it was initially a good idea of adding line breaks inside pre-formatted blocks...

the formatter by design applies high penalties to solutions which are not staying within `maxColumn`.

> UPD: I found out that there is [scalameta/scalafmt#3080](https://github.com/scalameta/scalafmt/pull/3080) already that you created which should help to avoid unexpected line breaks...

as you could see, the only solution was to apply a higher penalty for line breaks.

in this case, i would have increased `maxColumn` or rewrote the code so that it doesn't overflow. after all, exceeding the margins impedes readability, especially since you're using string constants to build code templates.

Reply:
> or rewrote the code so that it doesn't overflow.

I bet this should be the best option.